### DISCUSSION TITLE
What does mean "size of prediction is not equal to the size of instances" in batcher?

### DISCUSSION THREAD

**Discussion by hyunjoon1015** [kubeflow/kfserving#3540] [Q&A]
Hello guys,
I want to use batcher in my model prediction, so I'm following custom model serving guide ( https://kserve.github.io/website/0.9/modelserving/v1beta1/custom/custom_model/#create-your-custom-model-server-by-extending-model-class)

But when I add batcher in isvc yaml and try to send request, curl command gets response like,
```
{'message': 'size of prediction is not equal to the size of instances', 'batchId': 'd37f768e-0492-440a-bd46-2219365fef05', 'predictions': None}
```

In case of batcher agent container, it looks working fine.
The log of agent batcher was,
```
"caller": "batcher/handler.go:182", "message": "batch predict with size 3 /v1/models/batch:predict" ...
```

So, what does "size of prediction is not equal to the size of instances" mean??

**Comment by sivanantha321** [comment]
It means number of inputs not equal to the number of predictions. That is for example, if you provide three instances(or inputs), the prediction should contain three output values. During batching multiple inputs are combined an sent as a single request and if the prediction output does not match the input length, you will get this error.

For this 3 instances
```json
{
"instances": [
     [1, 2],
     [1, 2],
     [1, 2]
  ]
}
```
you should get 3 prediction like below, but if it does not match the input length you will get the error.

```json
{
"predictions": [1, 2, 3]
}
```
Note: This is just an example input format. The format may change depending on how the model consumes the input.
